installer/configurators/jenkins.py

```diff
@@ -0,0 +1,78 @@
+import subprocess
+import os
+import sys
+from pathlib import Path
+from installer.configurators.common import get_installer_root, get_tfvars_file, replace_tfvars
+
+jenkins_pem = get_installer_root() + "/jenkinskey.pem"
+
+
+def check_jenkins_pem():
+    """
+        Check if the user has provided jenkinskey.pem
+    """
+    print(
+        " Please make sure that you have the ssh login user names of jenkins and bitbucket servers."
+    )
+    # Check if user dropped the jenkins key into the installer root (scenario 1 only)
+    if not Path.is_file(jenkins_pem):
+        sys.exit(
+            "File jenkinskey.pem not found in installer root directory, kindly add and run the installer again! "
+        )
+
+    # Make sure the PEM the user dropped in has the right perms
+    subprocess.call('sudo chmod 400 {0}'.format(jenkins_pem))
+
+
+def check_jenkins_user(url, usernamepw):
+    """
+        Check if the jenkins user is present in Jenkins server
+    """
+    jenkins_url = 'http://' + url + ''
+    # Download the CLI jar from the jenkins server
+    subprocess.call([
+        'curl', '-sL', jenkins_url + '/jnlpJars/jenkins-cli.jar', '-o',
+        'jenkins-cli.jar'
+    ])
+
+    # Call the server and make sure user exists
+    cmd = [
+        'java', '-jar', 'jenkins-cli.jar', '-s', jenkins_url, 'who-am-i',
+        '--username', usernamepw[0], '--password', usernamepw[1]
+    ]
+    subprocess.call(
+        cmd, stdout=open("output", 'w'), stderr=open("output", 'w'))
+
+    if 'authenticated' in open('output').read():
+        os.remove('output')
+        return 1
+    else:
+        os.remove('output')
+        return 0
+
+
+def update_jenkins_terraform(ip, userpw, sshuser, ssh_port, secgrp, subnet):
+    replace_tfvars('jenkins_elb', ip, get_tfvars_file())
+    replace_tfvars('jenkins_security_group', secgrp, get_tfvars_file())
+    replace_tfvars('jenkins_subnet', subnet, get_tfvars_file())
+    replace_tfvars('jenkinsuser', userpw[0], get_tfvars_file())
+    replace_tfvars('jenkinspasswd', userpw[1], get_tfvars_file())
+    replace_tfvars('jenkins_ssh_login', sshuser, get_tfvars_file())
+    replace_tfvars('jenkins_ssh_port', ssh_port, get_tfvars_file())
+    replace_tfvars('jenkins_ssh_key', '{0}'.format(jenkins_pem), get_tfvars_file())
+
+
+def configure_jenkins(elb, userpw, ip, sshuser, secgrp, subnet):
+    # Check is the jenkins user exist in jenkins server
+    if not check_jenkins_user(elb, userpw):
+        sys.exit(
+            "Kindly provide an 'Admin' Jenkins user with correct password and run the installer again!"
+        )
+
+    check_jenkins_pem()
+
+    # Default Jenkins instance ssh Port
+    ssh_port = "22"
+
+    replace_tfvars('dockerizedJenkins', 'false', get_tfvars_file(), quoteVal=False)
+    update_jenkins_terraform(elb, userpw, ip, sshuser, ssh_port, secgrp, subnet)
```

Comment:
7 parameters, but the actual function has only 6. Also, we might need jenkins_public_ip to ssh. Elb might have different IP each time.

Reply:
So we don't even need an `elb` in this context, right? It's useless, we only want something we can connect to the Jenkins instance with.

We asked for both `elb` and public IP in the old installer, I confess I have no idea why we need both, just included both here (and was in the middle of removing one in favor of the other, which is why the args are jacked)

Reply:
Confirmed, `jenkins_public_ip` was removed because it was unused, we only need to ask for an `endpoint` which can be the public DNS or public IP of the Jenkins instance, don't care which.

This will be stored in the Terraform variable `jenkins_elb` because that's the variable most scripts reference but it doesn't have to be an ELB.

Reply:
Updated code to simply ask for a publicly accessible IP or a publicly resolvable DNS name, either should work.

Reply:
Ok. If user provides public IP only, then ssh will work. But the jenkins endpoint will have to pointed to some port like 8080.

Reply:
I'm not quite sure what you mean?

Yes, if something needs to hit a specific port, it will need to append that port to the IP.

Reply:
For example - Jenkins endpoint: 54.XXX.XXX.XX7:8080
In the chefProvision, it will connect through Jenkins END point using port 22, So it will become, 
54.XXX.XXX.XX7:8080:22

connection {
    host = "${lookup(var.jenkinsservermap, "jenkins_elb")}"
    user = "${lookup(var.jenkinsservermap, "jenkins_ssh_login")}"
    port = "${lookup(var.jenkinsservermap, "jenkins_ssh_port")}"
    type = "ssh"
    private_key = "${file("${lookup(var.jenkinsservermap, "jenkins_ssh_key")}")}"
  }

Reply:
Alright, that's scenario1 stuff, let me try and find all the spots in scenario1 where we're doing this and fix/verify.

Reply:
Do we use a nonstandard Jenkins port in our internal use of scenario1?

If not, then I'm gonna hardcode it to the default like we do the SSH port in the existing code, since that could also change and we don't bother with making it configurable.

Reply:
Alright, changed the installer to optionally accept a default port for Jenkins

`jenkins_rawendpoint` will be `<jenkins_endpoint>` and will be used for the SSH in the`chefProvisioner.tf` connection, since that's literally the only spot it's needed.

`jenkins_elb` will be `<jenkins_endpoint>:<jenkins_defaultport>` and will be used for everything else.

Reply:
ok, then check_jenkins_user(endpoint, userpw) should be appended with defaultport

Reply:
Good catch, done.